src/app/records/PresentationNodeLeaf.tsx

```diff
@@ -13,17 +15,103 @@ export default function PresentationNodeLeaf({
   ownedItemHashes,
   completedRecordsHidden,
   redactedRecordsRevealed,
+  sortRecordProgression,
 }: {
   node: DimPresentationNodeLeaf;
   ownedItemHashes?: Set<number>;
   completedRecordsHidden: boolean;
   redactedRecordsRevealed: boolean;
+  sortRecordProgression: boolean;
 }) {
+  const newNode = { ...node };
+  if (sortRecordProgression) {
+    /**
+     * RECORDS
+     */
+    if (node.records) {
+      newNode.records = orderBy(
+        newNode.records,
+        (record) => {
+          /**
+           * Triumph is already completed so move it to back of list.
+           * State is a bitmask where the ones place is for when the
+           * record has been redeemed. The other checks move completed
+           * but not redeemed to the back.
+           */
+          if (
+            record.recordComponent.state & DestinyRecordState.RecordRedeemed ||
+            record.recordComponent.state & DestinyRecordState.CanEquipTitle ||
+            !record.recordComponent.state
+          ) {
+            return -1;
+          }
+
+          // check which key is used to track progress
+          let objectives;
+          if (record.recordComponent.intervalObjectives) {
+            objectives = record.recordComponent.intervalObjectives;
+          } else if (record.recordComponent.objectives) {
+            objectives = record.recordComponent.objectives;
+          } else {
+            // its a legacy triumph so it has no objectives
+            return 0;
+          }
+
+          /**
+           * Iterate through the objectives array and return the average
+           * of its % progress.
+           */
+          let totalProgress = 0;
+          for (const x of objectives) {
+            // some progress bars exceed its completionValue
+            if (x.complete) {
+              totalProgress += 1;
+              continue;
+            }
+            totalProgress += x.progress! / x.completionValue;
```

Comment:
Objectives have lots of different meanings - see https://github.com/DestinyItemManager/DIM/blob/master/src/app/progress/Objective.tsx. Not all of them represent a completable progress bar.

Reply:
I did it this way initially as a blanket solution because I was having trouble distinguishing the different types of triumphs from each other. Are there some triumphs that would need to have their own logic? For example, at the start, I wasn't sure how I should deal with triumphs that have milestones since they share their progress.
![image](https://github.com/DestinyItemManager/DIM/assets/99226958/84b01093-5bf2-4506-8e45-26eecf57ce70)

 Are there triumphs that need to be addressed differently?